source/Engine.cpp

```diff
@@ -2194,7 +2194,9 @@ void Engine::DoCollisions(Projectile &projectile)
 		// For weapons with a trigger radius, check if any detectable object will set it off.
 		double triggerRadius = weapon.TriggerRadius();
 		if(triggerRadius)
-			for(const Body *body : shipCollisions.Circle(projectile.Position(), triggerRadius))
+		{
+			vector<Body *> circleCollisions;
```

Comment:
One of the benefits of the previous code is that the (cached) vectors are not re-allocated on every frame.

Reply:
I could do thread-local caching for those vectors if that's a problem. It's just often more effort than it is worth.

Reply:
No, I think it is better not to cache those vectors at all. But starting with a good initial capacity makes the code quite a bit faster.

Reply:
Simply calling the [reserve-function](https://cplusplus.com/reference/vector/vector/reserve/) could be sufficient.

Reply:
I reserved the ship count as the size on these vectors. I have a feeling it's waaaay more than what will be used, but I don't really know what a good default would be.

Reply:
I have put in some suggestions for default values.